presentation/src/main/java/com/example/nguyenthanhtungh/unsplashphoto/ui/search/SearchFragment.kt

```diff
@@ -0,0 +1,130 @@
+import com.example.nguyenthanhtungh.unsplashphoto.ui.search.adapter.CollectionAdapter
+import com.example.nguyenthanhtungh.unsplashphoto.ui.search.adapter.PhotoAdapter
+import com.example.nguyenthanhtungh.unsplashphoto.util.ITEM_DECORATION
+import com.example.nguyenthanhtungh.unsplashphoto.util.SPAN_COUNT
+import org.koin.android.viewmodel.ext.android.viewModel
+
+class SearchFragment : BaseFragment<FragmentSearchBinding, SearchViewModel>(),
+    SwipeRefreshLayout.OnRefreshListener {
+
+    companion object {
+        private const val QUERY_STRING = "QUERY_STRING"
+        const val TAG = "SearchFragment"
+        fun newInstance(query: String?) = SearchFragment().apply {
+            arguments = Bundle().apply {
+                putString(QUERY_STRING, query)
+            }
+        }
+    }
+
+    override val bindingVariable: Int = BR.searchViewModel
+
+    override val viewModel by viewModel<SearchViewModel>()
+
+    override val layoutId: Int = R.layout.fragment_search
+
+    override fun initComponent(viewDataBinding: FragmentSearchBinding) {
+        val query = arguments?.getString(QUERY_STRING) ?: return
+
+        val searchCollectionAdapter =
+            CollectionAdapter(
+                onItemClick = {
+                    goToCollectionDetailFragment(it)
+                }
+            )
+
+        val searchPhotoAdapter =
+            PhotoAdapter(
+                onItemClick = {
+                    goToDetailFragment(it)
+                }
+            )
+
+        viewDataBinding.onBackPress = View.OnClickListener {
+            onBackPress()
+        }
+
+        val endlessScrollListener = EndlessScrollListener { viewModel.onLoadMore(query) }
+        val decoration = RecyclerItemDecoration(ITEM_DECORATION)
+        viewDataBinding.apply {
+            recyclerPhoto.apply {
+                adapter = searchPhotoAdapter
+                layoutManager = GridLayoutManager(context, SPAN_COUNT)
+                addItemDecoration(decoration)
+                addOnScrollListener(endlessScrollListener)
+            }
+
+            recyclerCollection.apply {
+                adapter = searchCollectionAdapter
+                layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+                addItemDecoration(decoration)
+            }
+        }
+
+        viewBinding.searchSwipeLayout.setOnRefreshListener(this@SearchFragment)
+
+        viewModel.apply {
+
+            queryString.value = arguments?.getString(QUERY_STRING)
+
+            listSearchCollection.observe(this@SearchFragment, Observer {
+                searchCollectionAdapter.submitList(it)
+            })
+
+            listSearchPhotos.observe(this@SearchFragment, Observer {
+                searchPhotoAdapter.submitList(it)
+            })
+            firstLoad(query)
+
+            isLoadMore.observe(this@SearchFragment, Observer {
+                if (it == null) return@Observer
```

Comment:
chỗ này có thể viết thế này cũng cùng ý nghĩa: if (it ==true) endlessScrollListener.isLoading = it